docs/validation/validators.md

```diff
@@ -170,9 +209,49 @@ const date = {
   date: null // Valud missing, but this will be ignored.
 }
 
-hms(date, ['y', 'm']) // <- true
+requiredYmd(date, ['y', 'm']) // <- true
 ```
 
+## `requiredYmdIf`
+
+Checks if the given year, month and date is present when the condition passed is truthy. This validator will not validate if the values are valid numbers. To check the validity of the value, use [`ymd`](#ymd) validator.
+
+```ts
+import { type Ymd } from '@globalbrain/sefirot/lib/support/Day'
+import { type RequiredIfCondition } from '@globalbrain/sefirot/lib/validation/validators'
+
+function requiredYmd(
+  ymd: Ymd,
+  condition: RequiredIfCondition
+  required: ('y' | 'm' | 'd')[] = ['y', 'm', 'd']
+): boolean
+```
+
+```ts
+const date = {
+  year: 1985,
+  month: null, // Value missing.
+  date: 10
+}
+
+requiredYmdIf(date, true) // <- false
+requiredYmdIf(date, false) // <- true
+```
+
+You may pass 3rd argument to specify which fields to validate. For example,if you pass `['y', 'm']`, it will validate only year and month.
```

Comment:
```suggestion
You may pass 3rd argument to specify which fields to validate. For example, if you pass `['y', 'm']`, it will validate only year and month.
```

Reply:
Fixed: 46e2656